hermes/bin/src/vfs/mod.rs

```diff
@@ -2,13 +2,82 @@
 
 mod bootstrap;
 
+use std::io::{Read, Write};
+
 pub(crate) use bootstrap::VfsBootstrapper;
 
-use crate::hdf5::{self as hermes_hdf5};
+use crate::hdf5::{self as hermes_hdf5, Path};
 
 /// Hermes virtual file system type.
+#[derive(Clone, Debug)]
 pub(crate) struct Vfs {
     /// HDF5 root directory of the virtual file system.
     #[allow(dead_code)]
     root: hermes_hdf5::Dir,
+    // TODO: add permissions RWX
+}
+
+impl Vfs {
+    /// Reads in data in bytes, the number of which is specified by the caller,
+    /// from the hdf5 file and stores then into a buffer supplied by the calling process.
+    #[allow(dead_code)]
+    pub(crate) fn read(&self, path: Path) -> anyhow::Result<Vec<u8>, anyhow::Error> {
```

Comment:
I think we need to also have a fn that returns the File object so we can keep it open for reading/writing. That's something the WASI filesystem implementation is going to need.

I'm wondering if this PR will include that?

Reply:
we need to add permission too

Reply:
not in this one, just a basic scaffold in order to read to static files from runtime extension (for demo). We can iterate.